segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/SystemJournal.java

```diff
@@ -229,10 +230,23 @@ public void commitRecords(Collection<SystemJournalRecord> records) throws ChunkS
         }
         // Persist
         synchronized (lock) {
-            writeToJournal(bytes);
-            // Add a new log file if required.
-            if (!chunkStorage.supportsAppend() || !config.isAppendEnabled()) {
-                startNewJournalFile();
+            boolean done = false;
+            while (!done) {
+                try {
+                    writeToJournal(bytes);
+                    done = true;
+                } catch (ExecutionException e) {
```

Comment:
Why just `ExecutionException`? Can it be `Exception`, `CompletionException`? `Exceptions.unwrap` will handle all of those.

Reply:
I want to catch and handle a very specific exception InvalidOffsetException wrapped in ExecutionException.
Other exception should bubble up and loop should not be retried here.

Reply:
What if that exception is wrapped in CompletionException?